File: src/crew/dismissCrew.ts
import type { CrewMember } from "../data/types/crewMember.ts";
import type { DismissResult } from "../data/types/dismissResult.ts";

// Phase 4 GDD §2.7 -- voluntary dismissal, always succeeds for the
// crew member's actual owner.
export function dismissCrew(crewMember: CrewMember, playerId: string): DismissResult {
  if (crewMember.hiredByPlayerId !== playerId) {
    return { dismissed: false, reason: "player does not own this crew member" };
  }
  return { dismissed: true };
}

File: unity/ProfitableCore/Simulation/DismissCrew.cs
using Profitable.Core.Schema;

namespace Profitable.Core.Simulation;

// Ports src/crew/dismissCrew.ts. Voluntary dismissal, always succeeds for
// the crew member's actual owner.
public static class DismissCrewSimulation
{
    public static DismissResult DismissCrew(CrewMember crewMember, string playerId)
    {
        if (crewMember.HiredByPlayerId != playerId)
        {
            return new DismissResult { Dismissed = false, Reason = "player does not own this crew member" };
        }
        return new DismissResult { Dismissed = true };
    }
}
